app/cli/internal/telemetry/telemetry.go

```diff
@@ -49,28 +50,56 @@ func NewCommandTracker(client Client) *CommandTracker {
 
 // Track sends a command event to the telemetry.Client.
 func (t *CommandTracker) Track(ctx context.Context, cmd string, tags Tags) error {
+	// Ensure a valid client is available.
 	if t.client == nil {
 		return nil
 	}
 
-	computedTags := loadDefaultTags()
-	// Add on top of the default tags the ones passed as argument.
-	// TODO: Add a way to avoid the override default tags.
-	for k, v := range tags {
-		computedTags[k] = v
-	}
+	// Load default tags and merge with user-provided tags.
+	computedTags := mergeTags(loadDefaultTags(), tags)
+
+	// Set the command tag.
 	computedTags["command"] = cmd
 
-	var id string
-	id, err := machineid.ProtectedID("chainloop")
-	if err != nil {
-		// If the machine ID is not available, use a default ID otherwise the underlying library will fail.
-		id = "default-id"
-	}
+	// Determine the user ID.
+	id := determineUserID(computedTags)
 
+	// Track the event with computed tags.
 	return t.client.TrackEvent(ctx, commandTrackerEventName, id, computedTags)
 }
 
+// Merges two tag maps.
+func mergeTags(defaultTags, userTags map[string]string) map[string]string {
+	result := make(map[string]string)
+	for k, v := range defaultTags {
+		result[k] = v
+	}
+	for k, v := range userTags {
+		result[k] = v
+	}
+	return result
+}
+
+// Determines the user ID using available information or generates a random one.
+func determineUserID(tags map[string]string) string {
```

Comment:
it would be great to add some tests for this method to help explain what it does.